pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/geospatial/MutableH3Index.java

```diff
@@ -49,6 +53,17 @@ public MutableH3Index(H3IndexResolution resolution)
     _lowestResolution = resolution.getLowestResolution();
   }
 
+  @Override
+  public void add(@Nonnull Object value, int dictId, int docId) {
+    Geometry geometry = GeometrySerializer.deserialize((byte[]) value);
+    add(geometry);
+  }
+
+  @Override
+  public void add(@Nonnull Object[] values, @Nullable int[] dictIds, int docId) {
+    throw new UnsupportedOperationException("Mutable H3 indexes are not supported for multi-valued columns");
```

Comment:
In fact previous code what it does is to just ignore the value. Now the H3IndexType itself is the one that returns null (aka no mutable supplot) when the column is multi-valuated